public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -37,6 +41,94 @@ interface TimeAndValue {
   [TIME_SERIES_VALUE_FIELD_NAME]: number;
 }
 
+// V2 result trasnformer used to transform query results from queries that were run trough prometheus backend
+export function transformV2(response: DataQueryResponse, options: DataQueryRequest<PromQuery>) {
+  // Get refIds that have table format as we need to process those to table reuslts
+  const tableRefIds = options.targets.filter((target) => target.format === 'table').map((target) => target.refId);
+  const [tableResults, otherResults] = partition(response.data, (dataFrame) =>
+    dataFrame.refId ? tableRefIds.includes(dataFrame.refId) : false
+  );
+
+  // For table results, we need to transform data frames to table data frames
+  const tableFrames = tableResults.map((dataFrame) => {
+    const df = transformDFoTable(dataFrame, options.targets.length);
+    return df;
+  });
+
+  // Everything else is processed as time_series result and graph preferredVisualisationType
+  const otherFrames = otherResults.map((dataFrame) => {
+    const df = {
+      ...dataFrame,
+      meta: {
+        ...dataFrame.meta,
+        preferredVisualisationType: 'graph',
+      },
+    };
+    return df;
+  });
+
+  // Order by refIds
+  const orderedFrames = [...otherFrames, ...tableFrames].sort((df1, df2) => {
+    if (df1.refId && df2.refId) {
+      if (df1.refId < df2.refId) {
+        return -1;
+      }
+    }
+    return 1;
+  });
```

Comment:
just out of curiosity: why do we want to order them?

Reply:
Good question. So we get the frames ordered by refIds and I wanted to keep that order cause I thought that the order matter for table panel. But thinking about it/ testing it now, I think it is not needed because even if i miss-order them, they have still order in table. I'll remove this as it is not needed.

Reply:
Updated in https://github.com/grafana/grafana/commit/c74547581a97705c8fa3bf2681015475e668ba12